game/server_core.cpp

```diff
@@ -163,64 +163,65 @@ void ServerCore::process_input(InputPacket packet, short id)
             break;
         }
     }
+    PlayerObject* client_player = pWorld.findPlayer(id);
 
-    size_t sz = size_t(packet.events.size());
-    if (sz > 1)
-    {
-        int a = 0;
-    }
+    float SCALE = 1.0f; // TODO: Scale for what? Can use config.ini to replace this
+    int num_events = int(packet.events.size());
 
-    float scale = float(reader.GetReal("graphics", "player_movement_scale", 4.0)) / sz;
+    // why does scale depend on # of events?
+    float scale = float(reader.GetReal("graphics", "player_movement_scale", 4.0)) / num_events;
     glm::vec3 turndir;
 
     // Process input events
-    for (int j = 0; j < packet.events.size(); j++)
+    glm::vec3 dir = glm::vec3(0.0f, 0.0f, -1.0f);
+    for (int j = 0; j < num_events; j++)
     {
-        glm::vec3 dir;
         int event = packet.events[j];
-
+        bool jumping = false;
         switch (event)
         {
-        case MOVE_FORWARD:
-            dir = glm::vec3(0.0f, 0.0f, -1.0f);
-            break;
-        case MOVE_BACKWARD:
-            dir = glm::vec3(0.0f, 0.0f, 1.0f);
-            break;
-        case MOVE_LEFT:
-            dir = glm::vec3(-1.0f, 0.0f, 0.0f);
-            break;
-        case MOVE_RIGHT:
-            dir = glm::vec3(1.0f, 0.0f, 0.0f);
-            break;
-        case JUMP:
+            case MOVE_FORWARD:
+                dir = glm::vec3(0.0f, 0.0f, -1.0f);
+                client_player->move();
+                break;
+            case MOVE_BACKWARD:
+                dir = glm::vec3(0.0f, 0.0f, 1.0f);
+                client_player->move();
+                break;
+            case MOVE_LEFT:
+                dir = glm::vec3(-1.0f, 0.0f, 0.0f);
+                client_player->move();
+                break;
+            case MOVE_RIGHT:
+                dir = glm::vec3(1.0f, 0.0f, 0.0f);
+                client_player->move();
+                break;
+            case JUMP:
             {
-            dir = glm::vec3(0.0f, 1.0f, 0.0f);
-            glm::mat4 t = glm::translate(glm::mat4(1.0), dir * scale);
-            world = t * world;
-            continue;
-            break;
+                jumping = true;
+                if(client_player->getPosition().y == 0)
```

Comment:
not necessary for this pr, but noting that it'd probably be more appropriate to check for like a normal vertical force than for y=0 (in case players want to jump off furniture etc.)

Reply:
true will work on that